internal/namespaces/inference/v1beta1/custom_deployment.go

```diff
@@ -96,7 +96,7 @@ func deploymentCreateBuilder(c *core.Command) *core.Command {
 				publicEndpoint = nil
 			}
 			privateNetwork := &inference.EndpointSpecPrivateNetwork{}
-			if endpoint.EndpointSpec == nil {
+			if endpoint.EndpointSpec.PrivateNetwork == nil {
```

Comment:
What if `endpoint.EndpointSpec` is already nil?

Reply:
It creates a public endpoint.